lib/_http_incoming.js

```diff
@@ -121,7 +121,7 @@ IncomingMessage.prototype._addHeaderLines = function(headers, n) {
 // and drop the second. Extended header fields (those beginning with 'x-') are
 // always joined.
 IncomingMessage.prototype._addHeaderLine = function(field, value, dest) {
-  field = field.toLowerCase();
+  field = field.toLowerCase().trim();
```

Comment:
I think the `trim()` should be moved to the `default:` case to avoid unnecessary perf hits for the common header names.